### PR TITLE
Add collection:truncate

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

This PR implements the `collection:truncate` method with its unit tests.

### How should this be manually tested?

Clone this branch and run unit tests
`./gradlew test`

When it succeed, compile it

`./gradlew jar`

Initiate another java project by adding the compiled SDK as a dependency.

Then, run Kuzzle, create an index `nyc-open-data`, a `yellow-taxi` collection with several documents in it.
Finally, run this code

```java
import io.kuzzle.sdk.*;
import io.kuzzle.sdk.Options.KuzzleOptions;
import io.kuzzle.sdk.Options.Protocol.WebSocketOptions;
import io.kuzzle.sdk.Protocol.WebSocket;

import java.util.concurrent.ConcurrentHashMap;

public class truncateCollection {
    private static Kuzzle kuzzle;

    public static void main(String[] args) {
        WebSocketOptions opts = new WebSocketOptions();
        opts.setAutoReconnect(true).setConnectionTimeout(42000);

        try {
            WebSocket ws = new WebSocket("localhost", opts);

            kuzzle = new Kuzzle(ws, (KuzzleOptions) null);

            kuzzle.connect();

            kuzzle.getCollectionController().truncate("nyc-open-data", "yellow-taxi").get();
            
        }  catch (Exception e) {
            e.printStackTrace();
        }

        kuzzle.disconnect();
    }
};
```

### CODE REVIEW

File: doc/3/controllers/collection/truncate/index.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+code: true
+type: page
+title: truncate
+description: Remove all documents from collection
```
Comment (Aschen) [inline doc/3/controllers/collection/truncate/index.md]:
```suggestion
description: Remove all documents from a collection
```

File: doc/3/controllers/collection/truncate/index.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+# truncate
+
+Removes all documents from a collection, while keeping the associated mapping.
```
Comment (Aschen) [inline doc/3/controllers/collection/truncate/index.md]:
```suggestion
Removes all documents from a collection, while keeping the associated mappings.
```

File: doc/3/controllers/collection/truncate/index.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+<br/>
+
+| Arguments    | Type              | Description     |
+| ------------ | ----------------- | --------------- |
+| `index`      | <pre>String</pre> | Index name      |
+| `collection` | <pre>String</pre> | Collection name |
+
+## Returns
+
+A `ConcurrentHashMap` which has an `acknowledged` Boolean.
```
Comment (Aschen) [inline doc/3/controllers/collection/truncate/index.md]: IMHO, you can return nothing

Comment (Yoann-Abbes) [reply doc/3/controllers/collection/truncate/index.md]: The API returns it
https://docs.kuzzle.io/core/2/api/controllers/collection/truncate/

Comment (Aschen) [reply doc/3/controllers/collection/truncate/index.md]: Yes but it's useless IMHO

Comment (scottinet) [reply doc/3/controllers/collection/truncate/index.md]: The API returns an acknowledgement. This is a way of saying: "the query has been accepted and it ran fine, but there is no result to return".

So, in a SDK, this is a same than a function returning normally. Bothering with creating a ConcurrentHashMap just to return an acknowledgement has no point.

Comment (scottinet) [reply doc/3/controllers/collection/truncate/index.md]: And to me, this warrants a "requested change", so I'm going to block this PR over this issue.

File: src/main/java/io/kuzzle/sdk/API/Controllers/CollectionController.java [hunk 4]
```diff
@@ -6,6 +6,7 @@
 import io.kuzzle.sdk.Kuzzle;
 
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
```
Comment (scottinet) [inline src/main/java/io/kuzzle/sdk/API/Controllers/CollectionController.java]: This dependency seems unused

**Pull request merged**